### DISCUSSION TITLE
How BETA exactly is the Groups feature?

### DISCUSSION THREAD

**Discussion by bchanudet** [dani-garcia/vaultwarden#3384] [General]
Hello, 

To make the use of Vaultwarden easier in my company, I'd be interested into enabling the Groups feature on our instance. 

#246 lists the Group feature as implemented, but there is a warning about known issues. The wiki (and the config file) also indicates the feature as in Beta. Alas I haven't been able to find any real mentions of these issues in the Issues list. Also I've seen some other work being done recently around the groups feature like #3354. Therefore I'm a little bit confused about the status of the feature.

Is the Groups feature still considered as BETA? If so, what are the risks of enabling it in our instance? Could we "brick" the database in a way that makes all our passwords unreadable? 

I know nothing about Rust but I'd be willing to start learning it. Is there somewhere a list of those "known issues" that I could take a look at?

Many thanks for everything you do including the new version just released. Very solid work!

**Comment by AleksejEgorov** [comment]
Yep, it's requested feature, but I can't find description of these known issues. Are they described?

**Comment by BlackDex** [comment]
The main problem is that there are several SQL Queries which might have issues in regards to allowing access to ciphers which are allowed to be shown via groups. This needs some more **tlc** in my opinion, and maybe some de-normalization of some queries and schemas.

**Reply by bchanudet** [reply]
OK, SQL queries are my daily jam so I may be able to help on this part. :) I'll see if I can create a dev environment on my computer and have a look at this.

**Reply by BlackDex** [reply]
@bchanudet I'm a bit rusty with queries these days. Like 20 years ago I also did that daily.
So.... If you want. We would really appreciate someone to look at it.

We already have a nice feature added to output all the queries which are executed, which helped me in the past. If you need any more pointers please join us on our matrix chat channel, and ask there, or PM me on matrix :).
